kpatch-build/lookup.c

```diff
@@ -153,32 +164,46 @@ static int locals_match(struct lookup_table *table, int idx,
 	return 1;
 }
 
-static void find_local_syms(struct lookup_table *table, char *hint,
-			    struct sym_compare_type *child_locals)
+static void find_local_syms(struct lookup_table *table, struct symbol *file_sym,
+		struct list_head *sym_list)
 {
 	struct object_symbol *sym;
+	struct object_symbol *lookup_table_sym = NULL;
 	int i;
 
-	if (!child_locals)
-		return;
-
 	for_each_obj_symbol(i, sym, table) {
 		if (sym->type != STT_FILE)
 			continue;
-		if (strcmp(hint, sym->name))
+		if (strcmp(file_sym->name, sym->name))
 			continue;
-		if (!locals_match(table, i, child_locals))
+		if (!locals_match(table, i, file_sym, sym_list))
 			continue;
-		if (table->local_syms)
+		if (lookup_table_sym)
 			ERROR("found duplicate matches for %s local symbols in %s symbol table",
-			      hint, table->objname);
+			      file_sym->name, table->objname);
 
-		table->local_syms = sym;
+		lookup_table_sym = sym;
 	}
 
-	if (!table->local_syms)
+	if (!lookup_table_sym)
 		ERROR("couldn't find matching %s local symbols in %s symbol table",
-		      hint, table->objname);
+		      file_sym->name, table->objname);
+
+	list_for_each_entry_continue(file_sym, sym_list, list) {
+		if (file_sym->type == STT_FILE)
+			break;
+		file_sym->lookup_table_sym = lookup_table_sym;
+	}
+}
+
+static void link_local_symbols(struct lookup_table *table, struct kpatch_elf *kelf)
```

Comment:
`link_local_symbols()` makes it sound like something different than finding, when in fact it's just a loop around `find_local_syms()`.  Maybe call it `find_local_syms_multiple()` or so?  Also I think this function deserves a comment describing why it's doing this.  I assume it's for analyzing a bigger .o file like a built-in.o or vmlinux.o.

Similarly, the commit message needs more detail, as "support object-files built from multiple sources" didn't make sense to me until after I studied the code.  The commit message should also give more info about what problem is being solved and why.

Reply:
Agreed, will do.